include/RAJA/util/Span.hpp

```diff
@@ -164,6 +164,14 @@ RAJA_HOST_DEVICE RAJA_INLINE Span<IterType, IndexType> make_span(
   return Span<IterType, IndexType>(begin, size);
 }
 
+template <typename Iter>
+RAJA_INLINE auto make_span(Iter &iterable) ->
+Span<typename Iter::iterator, decltype(std::distance(std::begin(iterable), std::end(iterable)))>
```

Comment:
Since we're C++14 now, the trailing return type is optional.  Not a problem to have it there since this is what comes out anyway, but we no longer need to repeat them like this.